### ISSUE TITLE
DASH Manifest Containing HLS Content

### ISSUE THREAD

**Issue by navruzm** [gpac/gpac#3023]
I'm using GPAC to create a live stream from an RTMP source and output it as both DASH and HLS manifests. Here’s the command I’m using:

```
gpac -i rtmp://localhost:1935/app/testing:gpac:gfreg=ffdmx:#HLSPL=720p-$Type$.m3u8 -o tmp/files/csfvin4limd12dn187pg/live.mpd:gpac:dual:segdur=3:profile=live:dmode=dynamic:sseg=true:ll_preload_hint=false:ll_rend_rep=false:tpl=false:stl=true:seg_sync=yes:template=r7mf2g_$RepresentationID$$FS$$Number$$Init=init$
```

The `live.mpd` manifest file is unexpectedly containing HLS content, as indicated in the logs below. Additionally, GPAC outputs `MPD cannot be translated to M3U8` but still creates the manifest.

```
gpac - GPAC command line filter engine - version 2.5-DEV-rev772-ga23a3245-master
(c) 2000-2024 Telecom Paris distributed under LGPL v2.1+ - https://gpac.io
```

```
2024-10-31T11:25:54.946Z        INFO    segmenter/segmenter.go:68       Starting stream with args: "-i rtmp://localhost:1935/app/testing:gpac:gfreg=ffdmx:
#HLSPL=720p-$Type$.m3u8 -o tmp/files/csfvin4limd12dn187pg/live.mpd:gpac:dual:segdur=3:profile=live:dmode=dynamic:sseg=true:ll_preload_hint=false:ll_rend_rep=f
alse:tpl=false:stl=true:seg_sync=yes:template=r7mf2g_$RepresentationID$$FS$$Number$$Init=init$"                                                               
[Dasher] PID video1 config changed during active period, forcing period switch                                   
[M3U8] No segment state in representation, MPD cannot be translated to M3U8                                      
[Dasher] failed to write M3U8 file: Bad Parameter                                                                
[Dasher] Changing HLS target duration from 3 to 4, either increase the segment duration or re-encode the content 

...

2024-10-31T11:26:02.634Z        INFO    segmenter/playlist.go:222       MPD file: tmp/files/csfvin4limd12dn187pg/live.mpd                                     
2024-10-31T11:26:02.635Z        INFO    segmenter/playlist.go:223       MPD file content: "#EXTM3U\n#EXT-X-VERSION:6\n#EXT-X-INDEPENDENT-SEGMENTS\n\n#EXT-X-ST
REAM-INF:BANDWIDTH=1274628,CODECS=\"avc1.4D401F,mp4a.40.2\",RESOLUTION=1172x720,FRAME-RATE=25,AUDIO=\"audio\",CLOSED-CAPTIONS=NONE\n720p-video.m3u8\n\n#EXT-X-
MEDIA:TYPE=AUDIO,GROUP-ID=\"audio\",NAME=\"1\",AUTOSELECT=YES,URI=\"720p-audio.m3u8\",CHANNELS=\"2\"\n"                                                       
```

I’m running this on Ubuntu 22.04 and encountering this issue. However, on my Fedora development machine with the same GPAC version, I don’t see this issue.

- [x] I looked for a similar issue and couldn't find any.
- [x] I tried with the latest version of GPAC. Installers available at https://gpac.io/downloads/gpac-nightly-builds/
- [x] I give enough information for contributors to reproduce my issue (meaningful title, github labels, platform and compiler, command-line ...). I can share files anonymously with this dropbox: https://www.mediafire.com/filedrop/filedrop_hosted.php?drop=eec9e058a9486fe4e99c33021481d9e1826ca9dbc242a6cfaab0fe95da5e5d95

**Comment by jeanlf** [comment]
I cannot reproduce, this is likely related to the source. Could you rip a falty rtmp session (`gpac -i rtmp://... -o rip.gsf`) and upload the result ?

**Comment by navruzm** [comment]
Here is the file: [rip.zip](https://github.com/user-attachments/files/17662757/rip.zip)

**Comment by jeanlf** [comment]
Thanks for the file.

I cannot reproduce the MPD containing m3u8 with it and I have no clue how this could happen. Can you reproduce on your sustem with this test file ?

I however found out what was triggering the error, a period switch triggered before processing the first packet due to late signaling of pixel aspect ratio by the demuxer. Now fixed on master, can you give this a try ?

**Comment by navruzm** [comment]
> Now fixed on master, can you give this a try ?

Thanks, the fix is working.

> I cannot reproduce the MPD containing m3u8 with it and I have no clue how this could happen. Can you reproduce on your sustem with this test file ?

Can you try with this file? Also, I forgot to mention, this is happening only at the start. It is generating MPD correctly after a few seconds.

[rip2.zip](https://github.com/user-attachments/files/17675133/rip2.zip)

**Comment by jeanlf** [comment]
Hmm I cannot reproduce with this file either (OSX or ubuntu 22.04), the data written to the output file is the expected type (checked with asserts before writing).

Could you check the command line is as above (except for source) and with which gpac revision you experience this with the test file ?

**Comment by rbouqueau** [comment]
@navruzm What is the status for you on this?

**Comment by rbouqueau** [comment]
@mekhti Thanks. Is it possible to upload a sample so that we can reproduce?

PS: you can share files anonymously with this dropbox: https://www.mediafire.com/filedrop/filedrop_hosted.php?drop=eec9e058a9486fe4e99c33021481d9e1826ca9dbc242a6cfaab0fe95da5e5d95

**Comment by mekhti** [comment]
> [@mekhti](https://github.com/mekhti) Thanks. Is it possible to upload a sample so that we can reproduce?
> 
> PS: you can share files anonymously with this dropbox: https://www.mediafire.com/filedrop/filedrop_hosted.php?drop=eec9e058a9486fe4e99c33021481d9e1826ca9dbc242a6cfaab0fe95da5e5d95

Yes, but it's a huge (2.94 GiB) file. I have sent a download link to your email.

**Comment by mekhti** [comment]
Do you have any updates on this issue?

**Comment by rbouqueau** [comment]
Unfortunately I got zero time to investigate open-source issues these days. Thanks for the sample btw.

**Comment by rbouqueau** [comment]
Seems like the intermediate issue is solved, and the original one is not. @jeanlf had reviewed the code again (in June) and confirmed this should be impossible - so it must be another issue (e.g. wrong command-line/media types) that created that situation.

Any news here? We can probably only fix if we can reproduce.

**Comment by rbouqueau** [comment]
Last call to get news on this. Did anyone see it again ?
